### ISSUE TITLE
after the animation encoder (e.g. ffmpeg) fails, the animation framework itself fails internally in various ways while trying to report the error

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#9205]
### Bug report

**Bug summary**

While trying some LD_PRELOAD tricks, test_save_animation_smoketest failed for me with
```
lib/matplotlib/tests/test_animation.py:173: in test_save_animation_smoketest
    dpi=dpi, codec=codec)
lib/matplotlib/animation.py:1261: in save
    writer.grab_frame(**savefig_kwargs)
/usr/lib/python3.6/contextlib.py:88: in __exit__
    next(self.gen)
lib/matplotlib/animation.py:237: in saving
    self.finish()
    raise RuntimeError(msg)
E   RuntimeError: Error creating movie, return code: 127 Try setting mpl.verbose.set_level("helpful")
```

Well that's tricky to do within the tests which set classic mode, so let's instead patch
```
            try:
                stdout = [s.decode() for s in self._proc._stdout_buff]
                stderr = [s.decode() for s in self._proc._stderr_buff]
                verbose.report("MovieWriter.finish: stdout: %s" % stdout,
                               level='helpful')
                verbose.report("MovieWriter.finish: stderr: %s" % stderr,
                               level='helpful')
            except Exception as e:
                pass
```
to raise the error instead (pass -> raise).
I now get
```
lib/matplotlib/tests/test_animation.py:173: in test_save_animation_smoketest
    dpi=dpi, codec=codec)
lib/matplotlib/animation.py:1262: in save
    writer.grab_frame(**savefig_kwargs)
/usr/lib/python3.6/contextlib.py:88: in __exit__
    next(self.gen)
lib/matplotlib/animation.py:237: in saving
    self.finish()
lib/matplotlib/animation.py:577: in finish
    "\n" + str(self._proc._stderr_buff))
E   AttributeError: 'Popen' object has no attribute '_stdout_buff'
```

**Code for reproduction**

Apply the suggested patch and run pytest (suggested with --instafail (pytest-instafail) so that the traceback appears early, as test_animation is one of the first ones to be run).

**Actual outcome**

An exception is thrown while trying to generate an error message.  (The actual error was due to the fact that I tried to LD_PRELOAD the fixed version of libfreetype into matplotlib, but that also affected the subprocess -- so that error is not really relevant.)

**Expected outcome**

The actual error message gets displayed.

**Matplotlib version**
  * Operating System: Arch Linux
  * Matplotlib Version: master
  * Python Version: 3.6
  * Jupyter Version (if applicable): NA
  * Other Libraries: 

The use of a private subprocess API seem to have come in through https://github.com/matplotlib/matplotlib/pull/5628.  We should definitely just use a public API here.  The private API still exists on Windows, but has been removed on Unices in https://github.com/python/cpython/commit/3a4586a (I think), i.e. all supported Python versions.

Probably never caught because no one actually uses verbose mode... (perhaps tests should use verbose mode by default, actually -- especially as pytest will buffer away the output unless a failure actually occurs).

**Comment by anntzer** [comment]
In the same vein: when saving an image fails, the animation framework tries to `communicate()` twice with the process, causing an additional exception on the way...

e.g. add a chmod+x file named `ffmpeg` to your PATH with the contents
```
#!/bin/sh
echo oops
return 1
```
and try running examples/animation/simple_anim.py after uncommenting the `anim.save` line.  You get the following exception:
```
Traceback (most recent call last):
  File "simple_anim.py", line 33, in <module>
    ani.save("movie.mp4")
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/animation.py", line 1194, in save
    writer.grab_frame(**savefig_kwargs)
  File "/usr/lib/python3.6/contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/animation.py", line 239, in saving
    self.finish()
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/animation.py", line 365, in finish
    self.cleanup()
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/animation.py", line 402, in cleanup
    out, err = self._proc.communicate()
  File "/usr/lib/python3.6/subprocess.py", line 843, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
  File "/usr/lib/python3.6/subprocess.py", line 1505, in _communicate
    selector.register(self.stdout, selectors.EVENT_READ)
  File "/usr/lib/python3.6/selectors.py", line 351, in register
    key = super().register(fileobj, events, data)
  File "/usr/lib/python3.6/selectors.py", line 237, in register
    key = SelectorKey(fileobj, self._fileobj_lookup(fileobj), events, data)
  File "/usr/lib/python3.6/selectors.py", line 224, in _fileobj_lookup
    return _fileobj_to_fd(fileobj)
  File "/usr/lib/python3.6/selectors.py", line 39, in _fileobj_to_fd
    "{!r}".format(fileobj)) from None
ValueError: Invalid file object: <_io.BufferedReader name=14>
```

**Comment by anntzer** [comment]
Closed by #12368 & #12369.
